pkg/reconciler/route/reconcile_resources.go

```diff
@@ -241,21 +241,7 @@ func (c *Reconciler) reconcileTargetRevisions(ctx context.Context, t *traffic.Co
 	return eg.Wait()
 }
 
-func (c *Reconciler) reconcileCertificates(ctx context.Context, r *v1alpha1.Route, desiredCerts []*netv1alpha1.Certificate) ([]*netv1alpha1.Certificate, error) {
-	result := []*netv1alpha1.Certificate{}
-	gvk := netv1alpha1.SchemeGroupVersion.WithKind("Certificate")
-	for _, desiredCert := range desiredCerts {
-		c.tracker.Track(objectRef(desiredCert, gvk), r)
-		cert, err := c.reconcileCert(ctx, r, desiredCert)
-		if err != nil {
-			return nil, err
-		}
-		result = append(result, cert)
-	}
-	return result, nil
-}
-
-func (c *Reconciler) reconcileCert(ctx context.Context, r *v1alpha1.Route, desiredCert *netv1alpha1.Certificate) (*netv1alpha1.Certificate, error) {
+func (c *Reconciler) reconcileCertificate(ctx context.Context, r *v1alpha1.Route, desiredCert *netv1alpha1.Certificate) (*netv1alpha1.Certificate, error) {
```

Comment:
We should have these per-"tag" too.

Reply:
I'm fine with following up, but please open an issue.

Reply:
Currently the certificate supports the DNS names for both the main route and all tag routes. See https://github.com/knative/serving/pull/3902/files/5563bc6b3cee385b88e0bc4374b4ed55e6656f90#diff-05400f68e9479b6fc5caaf943279559eR304

Do you mean splitting this single certificate to multiple certificates, and each of the certificate corresponds to one main/tag route?

Reply:
That's interesting.  I wonder if that will be higher or lower churn. 🤔

Reply:
Good question. Having multiple domains within one cert enforces a requirement that the Certificate provision solution (like `cert-manager) needs to support SAN, which is probably strong. I created an issue https://github.com/knative/serving/issues/4004 to track this problem and would like to follow up in a separate PR if you are OK with it.

While the slight benefit of having multiple domains within single cert is that it could save some LetsEncrypt quota when firstly starting a route with tag routes.